src/middleware/router.js

```diff
@@ -220,15 +220,23 @@ function sendRequestToRoutes (ctx, routes, next) {
             logger.info(`executing primary route : ${route.name}`)
             if (response.headers != null && response.headers['content-type'] != null && response.headers['content-type'].indexOf('application/json+openhim') > -1) {
               // handle mediator reponse
-              const responseObj = JSON.parse(response.body)
-              ctx.mediatorResponse = responseObj
+              let payload = ''
+              response.body.on('data',function(data){
```

Comment:
Looks good to me. 
Maybe just use lambda instead of function callback for consistency.

Reply:
Addressed in 1867f6f